controllers/openshift_controller.go

```diff
@@ -1828,31 +1682,15 @@ func (r *KataConfigOpenShiftReconciler) updateStatus() error {
 		return len(nodes.Items)
 	}()
 
-	r.kataConfig.Status.KataNodes.NodeCount = r.kataConfig.Status.TotalNodesCount
-
 	for _, node := range nodeList.Items {
-		if annotation, ok := node.Annotations["machineconfiguration.openshift.io/state"]; ok {
-
-			r.putNodeOnStatusList(&node)
-
-			switch annotation {
-			case NodeDone:
-				e := r.processDoneNode(&node)
-				if e != nil {
-					err = e
-				}
-			case NodeDegraded:
-			case NodeWorking:
-				e := r.processTransitioningNode(&node)
-				if e != nil {
-					err = e
-				}
-			default:
-				err = fmt.Errorf("Unexpected machineconfiguration.openshift.io/state: %v ", annotation)
-				r.Log.Info("Unexpected machineconfiguration.openshift.io/state", "node", node.GetName(), "state", annotation)
-			}
+		e := r.putNodeOnStatusList(&node)
+		if e != nil {
+			err = e
```

Comment:
In case of multiple errors, we're keeping only the last one right?
Can we at least log the error here before we potentially lose it?
Or as an alternative: if we meet an error, is it useful to continue the loop?

Reply:
Yes, a comment at the top of the functions make that point.

As for error handling, yes, my idea was that it's still useful to continue. The way I see it, if there's an error our status will always end up imprecise for the moment but it would be even more imprecise if we broke off on the first error.  That said, I'm not sure at the moment how rigorous we should be here, I'm guessing some practical experience will help.  IIRC the legacy status update behaved the same way and I don't remember any practical problems with that.

Errors that can come up here will likely tend to be transient.

Reply:
Sorry, I missed the comment at the top.
Thanks for the answer. Makes sense